docs/testing_image.md

```diff
@@ -0,0 +1,254 @@
+
+## verify image
+
+This image will run one or more verify scripts using the `hack/verify`. This
+is analogous to `kubernetes-verify`.
+
+The myriad of verify-* scripts in `hack` will move to `hack/verify-scripts/`.
+We will copy `hack/verify-all.sh` to `hack/verify` and make it accept arguments
+about which verifications to perform.
+Then we will make `hack/verify-all.sh` a wrapper to the new script.
+Then we will deprecate the old script. Then delete it.
+
+We will also delete `hack/jenkins/verify.sh` and
+`hack/jenkins/verify-dockerized.sh`
+
+
+### verify job-config
+
+This will control:
+
+* the verify subchecks to run
+
+The image will default to run all verify checks.
+
+
+## unit image
+
+This image will run unit tests using the `hack/test-go.sh` script. This is
+analgous to `kubernetes-test-go`.
+
+This script will replace the cacophony of similar scripts:
+* `hack/jenkins/gotest-dockerized.sh`
+* `hack/jenkins/gotest.sh`
+* `hack/jenkins/test-dockerized.sh`
+* `hack/test-go.sh`
+
+Any necessary logic will move into `hack/test-go.sh`.
+In particular the kubectl unit tests at `hack/test-cmd.sh` will
+migrate to a format that produce junit results.
+
+
+### unit job-config
+
+This will control:
+
+* the set of api versions to test
+* whether to include coverage data
+* how many tests to run in parallel
+* toggle race testing
+
+This will default to match what happens on PR jenkins.
+
+
+## integration image
+
+This image will run integration tests using the `hack/test-integration.sh`. This
+script is analogous to a subset of `kubernetes-test-go`.
+
+This will decouple the integration tests from `hack/test-go.sh`. Alternatively
+we could combine all the test-\* images into a single unit image. However the
+advantage of moving the integration tests into their own image is to ensure that
+unit tests never depend on a real etcd server.
+
+
+### integration job-config
+
+This will control:
+
+* level of concurrency
+* log level
+* api versions
+
+This will default to match what happens on PR jenkins.
+
+
+## e2e image
+
+This image will run e2e tests that deploy and test clusters on various
+platforms via `hack/e2e.go`. This is analogous to the `kubernetes-e2e-*` jobs.
+
+This script will eventually eliminate any intermediate scripts:
+* `hack/e2e-internal/*` in particular
+
+
+### e2e job-config
+
+This will control:
+
+* platform
+* concurrency
+* min startup pods
+* fail on resource leak toggle
+* ginkgo focus
+* ginkgo skip
+* node configuration (number, shape, disk size
+* any other options.
+
+This will default to match what happens on PR jenkins.
+
+
+## node\_e2e image
+
+This image will run node e2e tests via a new `hack/node_e2e.go` script. This is
+analogous to the `*-gce-e2e-ci` and `*-dockercanarybuild-ci` jobs.
+
+This script will move the logic to run tests that currently lives in
+[node-e2e.yaml] and into a new program that performs these actions instead.
+
+
+### node\_e2e job-config
+
+
+This will control:
+
+* git basedir
```

Comment:
pretty sure this is only relevant to how Jenkins checks out the repo. I don't think it needs to be in the image.